### PR TITLE
[frontend] use dynamic typing for useLogicalFiltersFromStorage (#1673)

### PR DESCRIPTION
**Pull request by Antoine-Quintin**
# Context
> Use TypeScript discriminated unions to provide exact required parameters for useLogicalFiltersFromStorage based on the service slug, instead of using optional parameters for all slugs

## Related issues

- #1673

# Checklist

## Quality

- [ ] I consider this work complete and ready for review
- [ ] I tested all features and edge cases
- [ ] I refactored code where necessary to improve quality
- [ ] I made sure my code meets development guidelines
- [ ] I performed a self-review of my code

## Testing

- [ ] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [ ] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-front/app/(public)/cybersecurity-solutions/[slug]/public-document-list-page-loader.tsx [hunk 1]
```diff
@@ -1,31 +1,34 @@
 'use client';
-import { useServiceListLocalStorage } from '@/components/service/components/use-service-list-local-storage';
-import { PublicDocumentListQuery } from '@/components/service/document/public-document.graphql';
+import {useServiceListLocalStorage} from '@/components/service/components/use-service-list-local-storage';
```
Comment (Kroustille) [inline apps/portal-front/app/(public)/cybersecurity-solutions/[slug]/public-document-list-page-loader.tsx]: You have issues in your linter here too

File: apps/portal-front/src/components/service/document/use-logical-filters-from-storage.tsx [hunk 2]
```diff
@@ -5,21 +5,34 @@ import { FilterKeyEnum } from '@generated/models/FilterKey.enum';
 import { LogicalOperatorEnum } from '@generated/models/LogicalOperator.enum';
 import { useMemo } from 'react';
 
+type SimpleFiltersParams = {
+  serviceInstanceSlug: ServiceSlug.OPEN_CTI_CUSTOM_DASHBOARDS | ServiceSlug.OPEN_AEV_SCENARIOS;
+  labels: LogicalMultiSelectSelection;
+  deployable?: never;
+  verified?: never;
+  integrationTypes?: never;
+  productVersions?: never;
```
Comment (Kroustille) [inline apps/portal-front/src/components/service/document/use-logical-filters-from-storage.tsx]:
```suggestion
type SimpleFiltersParams = {
  serviceInstanceSlug: ServiceSlug.OPEN_CTI_CUSTOM_DASHBOARDS | ServiceSlug.OPEN_AEV_SCENARIOS;
  labels: LogicalMultiSelectSelection;
```

File: apps/portal-front/src/components/service/document/use-logical-filters-from-storage.tsx [hunk 3]
```diff
@@ -5,91 +5,86 @@ import { FilterKeyEnum } from '@generated/models/FilterKey.enum';
...
-    deployable,
-    verified,
-    serviceInstanceSlug,
-    productVersions,
+    params.serviceInstanceSlug,
+    params.labels,
+    'deployable' in params ? params.deployable : undefined,
+    'verified' in params ? params.verified : undefined,
+    'integrationTypes' in params ? params.integrationTypes : undefined,
+    'productVersions' in params ? params.productVersions : undefined,
```
Comment (Kroustille) [inline apps/portal-front/src/components/service/document/use-logical-filters-from-storage.tsx]: We could use only one dependency:
```suggestion
    params
```

**Review by Kroustille — APPROVED** [review]
